gui/src/app/SPAnalysis/SPAnalysisContextProvider.tsx

```diff
@@ -12,12 +14,113 @@ type SPAnalysisContextProviderProps = {
 
 export const SPAnalysisContext = createContext<SPAnalysisContextType>({
     data: initialDataModel,
-    update: () => {}
+    update: () => { }
 })
 
-const SPAnalysisContextProvider: FunctionComponent<PropsWithChildren<SPAnalysisContextProviderProps>> = ({children}) => {
+
+enum QueryParamKeys {
+    StanFile = "stan",
+    DataFile = "data",
+    SamplingOpts = "sampling_opts",
+    Title = "title",
+    SONumChains = 'num_chains',
+    SONumWarmup = 'num_warmup',
+    SONumSamples = 'num_samples',
+    SOInitRadius = 'init_radius',
+    SOSeed = 'seed',
+}
+
+type QueryParams = {
+    [key in QueryParamKeys]: string | null
+}
+
+const fromSearchParams = (searchParams: URLSearchParams) => {
+    const query: QueryParams = {
+        stan: searchParams.get(QueryParamKeys.StanFile),
+        data: searchParams.get(QueryParamKeys.DataFile),
+        sampling_opts: searchParams.get(QueryParamKeys.SamplingOpts),
+        title: searchParams.get(QueryParamKeys.Title),
+        num_chains: searchParams.get(QueryParamKeys.SONumChains),
+        num_warmup: searchParams.get(QueryParamKeys.SONumWarmup),
+        num_samples: searchParams.get(QueryParamKeys.SONumSamples),
+        init_radius: searchParams.get(QueryParamKeys.SOInitRadius),
+        seed: searchParams.get(QueryParamKeys.SOSeed),
+    }
+    return query
+
+}
+
+
+const tryFetch = async (url: string) => {
+    console.log('Fetching content from', url);
+    try {
+        const req = await fetch(url);
+        if (!req.ok) {
+            console.error('Failed to fetch from url', url, req.status, req.statusText);
+            return undefined;
+        }
+        return await req.text();
+    } catch (err) {
+        console.error('Failed to fetch from url', url, err);
+        return undefined;
+    }
+}
+
+const deepCopy = (obj: any) => {
+    return JSON.parse(JSON.stringify(obj))
+}
+
+const fetchRemoteAnalysis = async (query: QueryParams) => {
+
+    // any special 'project' query could be loaded here at the top
+    const data: SPAnalysisDataModel = deepCopy(initialDataModel);
+
+    if (query.stan) {
```

Comment:
We could save this for later, but I think it would be better to iterate over the query parameters in this kind of fashion

```
for key in query
  if key == QueryParamKeys.StanFile
    data.stan = ...
  elif key == ...
    ..
  else
    print a warning that the query parameter was unused
```

The advantages: we'd get the warnings; we'd make use of the QueryParamKeys enum
```

Reply:
Good idea. I added a warning inside the `fromSearchParams` function (just to the console for now), since by the time we have this `query` object it would only ever have keys we knew about